package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,345 @@
+    structure assignment. One-hot encoding corresponds to C3 notation:
+    '-', 'H', 'E' are loop, helix and sheet, respectively. Input array must
+    have its last axis of shape 3: (n_residues, 3) or (n_frames, n_residues, 3)
+
+    Examples
+    --------
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> import numpy as np
+    >>> # encoding 'HE-'
+    >>> onehot = np.array([
+        [False, True, False],  # 'H'
+        [False, False, True],  # 'E'
+        [True, False, False]]) # '-'
+    >>> ''.join(translate(onehot))
+    'HE-'
+
+    Parameters
+    ----------
+    onehot : np.ndarray
+        input array of one-hot encoding in ('-', 'H', 'E') order
+
+    Returns
+    -------
+    np.ndarray
+        array of '-', 'H' and 'E' symbols with secondary structure
+
+    .. versionadded:: 2.8.0
+    """
+    C3_ALPHABET = np.array(["-", "H", "E"])
+    index = np.argmax(onehot, axis=-1)
+    return C3_ALPHABET[index]
+
+
+class DSSP(AnalysisBase):
+    """Assign secondary structure using DSSP algorithm.
+
+    Analyze a selection containing a protein and assign secondary structure
+    using the Kabsch-Sander algorithm [Kabsch1983]. Only a subset of secondary
+    structure categories are implemented:
+
+    - 'H' represents a generic helix (α-helix, π-helix or 3-<sub>10</sub> helix)
+    - 'E' represents 'extended strand', participating in beta-ladder (parallel
+      or antiparallel)
+    - '-' represents unordered part
+
+    The implementation was taken from the pydssp package (v. 0.9.0)
+    https://github.com/ShintaroMinami/PyDSSP by Shintaro Minami.
+
+    Parameters
+    ----------
+    u : Universe
+        input universe
+    guess_hydrogens : bool, optional
+        whether you want to guess hydrogens positions, by default ``True``.
+        Guessing is made assuming perfect 120 degrees for all bonds that N
+        atom makes, and a N-H bond length of 1.01 A.
+        If ``guess_hydrogens`` is False, hydrogen atom positions on N atoms
+        will be parsed from the trajectory, except for the "hydrogen" atom
+        positions on PRO residues, and an N-terminal residue.
+
+    Raises
+    ------
+    ValueError
+        if ``guess_hydrogens`` is True but some non-PRO hydrogens are missing.
+
+    Examples
+    --------
+
+    For example, you can assign secondary structure for a single PDB file:
+
+    >>> from MDAnalysis.analysis.dssp import DSSP
+    >>> from MDAnalysisTests.datafiles import PDB
+    >>> import MDAnalysis as mda
+    >>> u = mda.Universe(PDB)
+    >>> run = DSSP(u).run()
+    >>> print("".join(run.results.dssp[0][:20]))
+    '--EEEEE-----HHHHHHHH'
+
+    Also, per-frame dssp assignment allows you to build average
+    secondary structure -- ``DSSP.results.dssp_ndarray`` holds
+    (n_frames, n_residues, 3) shape ndarray with one-hot encoding of
+    loop, helix and sheet, respectively:
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> u = mda.Universe(...)
```

Comment:
please fill in ... (since we are trying to make sure code in docs can run)

Reply:
Good point, see #3925 — essentially, make sure that
```
cd package/docs/sphinx
# make doctest   ## add --keep-going to line 130: gives this command:
sphinx-build -v -W -b doctest --keep-going   source ../doctest
```
works for your file (I write output to a file `2>&1 | tee doctest.log` and look through the log.)

Reply:
Thanks, I tried to make as many as possible tests run (I believe there are no dssp-related failures anymore).